### PR TITLE
Add segment_variance and segment_normalize to utils.py

### PR DESCRIPTION
**Pull request by sooheon**
segment_normalize is an alternative to segment_softmax for normalizing attention weights across incoming edges.

Add tests

### CODE REVIEW

**Comment by sooheon** [comment]
Motivated by [Richter and Wattenhofer, (2020)](https://arxiv.org/abs/2005.09561), which uses normalization in place of softmax to good effect.

**Comment by sooheon** [comment]
I changed arg names of `segment_mean` to match `segment_softmax` and other segment fns from jax, but I realize that may be an annoying change, let me know if you want that reverted or have any other feedback.

File: jraph/_src/utils.py [hunk 1]
```diff
@@ -15,14 +15,17 @@
 """Utilities for working with `GraphsTuple`s."""
 
 import functools
-from typing import Any, Callable, Iterable, List, Mapping, Optional, Sequence, Union
+from typing import Any, Callable, Iterable, List, Mapping, Optional, Sequence, \
+  Union
 
 import jax
 import jax.numpy as jnp
+import jax.ops
```
Comment (jg8610) [inline jraph/_src/utils.py:23]: I don't think this is needed, can't you directly call jax.ops.segment_sum etc.?

File: jraph/_src/utils.py [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
 import jax.tree_util as tree
-from jraph._src import graph as gn_graph
 import numpy as np
+from jax import lax
 
+from jraph._src import graph as gn_graph
```
Comment (jg8610) [inline jraph/_src/utils.py:28]: Why have you changed the ordering?

**Review by jg8610 — COMMENTED** [review]
Thank you! This looks great, some minor comments.

Have you signed the CLA?

File: jraph/_src/utils.py (hunk 1 above)
Comment (sooheon) [reply jraph/_src/utils.py:23]: You can, but jax/__init__.py doesn't explicitly import jax.ops, it looks like accident of dependency resolution that it's available so editors will give a warning.

File: jraph/_src/utils.py (hunk 2 above)
Comment (sooheon) [reply jraph/_src/utils.py:28]: No meaning, the editor I use "cleans up" imports automatically. Will revert.

**Comment by sooheon** [comment]
Yep I have signed the CLA

**Comment by sooheon** [comment]
By the way, do you have any performance tips regarding implementation of the scatter functions?

**Comment by jg8610** [comment]
Thanks! I've merged.

What performance concerns do you have specifically? Some accelerators are more efficient that others with scatters/gathers (e.g. TPUs are less efficient than others). Your mileage may also vary depending on how XLA will optimize your jitted code - it may be that they are fused.

If there is something specific, I can take a look.
